addons/stock/models/product_strategy.py

```diff
@@ -63,6 +64,35 @@ def _default_product_id(self):
     package_type_ids = fields.Many2many('stock.package.type', string='Package Type', check_company=True)
     storage_category_id = fields.Many2one('stock.storage.category', 'Storage Category', ondelete='cascade', check_company=True)
     active = fields.Boolean('Active', default=True)
+    sublocation = fields.Selection([
+        ('no', 'No'),
+        ('last_used', 'Last Used'),
+        ('closest_location', 'Closest Location')
+    ], default='no')
+
+    @api.onchange('sublocation', 'location_out_id', 'storage_category_id')
+    def _onchange_sublocation(self):
```

Comment:
avoid onchange. Instead try to do compute with depends. It will avoid to have different behavior in front and backend. (onchange won't trigger with a write in the code while compute does)